### DISCUSSION TITLE
Diarization not working

### DISCUSSION THREAD

**Discussion by jgeddes-receptiviti** [deepgram/community#584] [General help] [answered]
### Which Deepgram product are you using?

Deepgram API

### Details

using `"diarize": "true"` does not produce a transcript with any speaker labels. Have tried many variations of the code. Simply getting a wall of text with all the speakers mixed in. (Note, file is too big when zipped to upload - though Assembly AI's API was able to diarize this exact file with fair accuracy).

### If you are making a request to the Deepgram API, what is the full Deepgram URL you are making a request to?

https://api.deepgram.com/v1/listen

### If you are making a request to the Deepgram API and have a request ID, please paste it below:

dd98d562-06b8-4323-87e4-0cf504bb1235

### If possible, please attach your code or paste it into the text box.

```python
import os
import logging
from datetime import datetime, timedelta
import requests
import json

# Set your Deepgram API key here
DEEPGRAM_API_KEY = "<>"

AUDIO_FILE = "/my-audio.wav"
def main():
    try:
        # STEP 1: Prepare the headers with the API key
        headers = {
            "Authorization": f"Token {DEEPGRAM_API_KEY}",
            "Content-Type": "audio/wav",
        }

        # STEP 2: Read the audio file
        with open(AUDIO_FILE, "rb") as file:
            audio_data = file.read()

        # STEP 3: Prepare the request payload
        payload = audio_data

        # STEP 4: Prepare the transcription options
        params = {
            "diarize": "true",
            "model": "nova-2"
        }

        # STEP 5: Make the API request
        before = datetime.now()
        response = requests.post("https://api.deepgram.com/v1/listen", headers=headers, params=params, data=payload)
        after = datetime.now()

        # Parse the response as JSON
        response_json = response.json()

        # Print the JSON response
        print(response_json)
        print("")

        difference = after - before
        print(f"time: {difference.seconds}")

        # Print the JSON response in a pretty format
        print(json.dumps(response_json, indent=4))
        print("")

    except Exception as e:
        print(f"Exception: {e}")
        

if __name__ == "__main__":
    main()
```

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by jpvajda** [comment]
@jgeddes-receptiviti 👋  something you can do to troubleshoot is trying to run this same request in our [API Playground](https://playground.deepgram.com/?diarize=true&language=en&model=nova-2) and see if the results differ

Are you using our Python SDK by any chance?

Also you said the file was fairly large, so I'm curious if you were able to try to split the file up if that results in a better outcome.

**Reply by writeonlymemory** [reply]
FWIW I am seeing the same thing even in the API playground.  Two very different voices (male and female) but all coming back as speaker 0. Has something changed recently perhaps?

[and FYI I was seeing the same thing with live transcription from Go but verified the behaviour in the playground with a recording]

**Reply by writeonlymemory** [reply]
Playing around with your sample audio (`https://static.deepgram.com/examples/en_NatGen_CallCenter_BethTom_CancelPhonePlan.wav`) I did get it to diarize - I guess the segments I was working with were too short. It seems to need longer examples of both speakers before it starts working.

**Comment by flexchar** [comment]
I've tried Deepgram too. First of all, it's speed is mind blowing. But diarization always returns a speaker 0. So I am hoping to try again in 2025 and see two speakers✌️

Also tried in Assembly AI's playground. It worked but wasn't accurate enough. 

I also tried Gemini 1.5 Pro. Produced by far the best results but it hallucinates too much of things not said 🤣

**Reply by imasha-ar** [reply]
Change the model and try again if you're using general nova-2 model try with other models like nova-2-phonecall,nova-2-medical,nova-2-meeting. https://developers.deepgram.com/docs/models-languages-overview

**Reply by Abhishek21k** [reply]
they still have this issue going on

**Comment by rensvis** [comment]
My issue was that the speaker field wasn't showing up at all with nova-2 streaming. When I switched to nova-2-meeting I started getting speaker results with different id's.

**Comment by avalsonigara15** [comment]
I tried Diarization on Deepgram Dashboard using the nova 3 model but sill have the issue, where it gives only `speaker_0` and then combines all the transcript together. Below are the snapshots of  the dashboard.
<img width="1665" height="1020" alt="image" src="https://github.com/user-attachments/assets/2df381a0-4d75-4573-a5f3-be3c439042b6" />
<img width="977" height="825" alt="image" src="https://github.com/user-attachments/assets/0edbc621-3a2f-4b0b-8137-9937c8785ecc" />

**Reply by camiloescobarg** [reply]
I'm getting the same as [avalsonigara15](https://github.com/avalsonigara15). Are there any news about this? thanks!

**Comment by naomi-deepgram** [comment, marked as answer]
Hey friends! For anyone experiencing this issue, please do me a favour.

I need the following from you:

1. Project ID
2. Request ID
3. The audio you transcribed (if this is not public, you can email it to me at `naomi.carrigan@deepgram.com`
4. The code you are running that triggers the issue (for those who see it on playground, screenshot the configs)

This will help me dive in and sus out what's goin on~
